src/commands/purge-key.test.js

```diff
@@ -0,0 +1,101 @@
+const should = require(`should`);
+const sinon = require(`sinon`);
+require(`should-sinon`);
+
+const util = require(`./../util`);
+const testData = require(`./../test.data`)
+const testSubject = require(`./purge-key`);
+
+const testCacheKey = `5812587d-1407-4a60-930f-ce39e2b7e6f7`;
+
+// Scope vars that we’ll redefine on each test.
+let options, purgeKeyStub, softPurgeKeyStub, Fastly;
+
+// Begin testing.
+describe(`purge-key`, () => {
+  // Before and after each test, we reset some stuff
+  // to ensure a fresh set of arguments. See .beforeEach()
+  // and .afterEach() near the bottom of this file.
+
+  it(`should stop without an API key`, (done) => {
+    delete options.apikey;
+    testSubject(options, Fastly, util, testCacheKey);
+    util.apiKeyPresent.should.be.calledOnce();
+    Fastly.should.not.be.called();
+    done();
+  });
+
+  it(`should stop without a Service ID`, (done) => {
+    delete options.serviceid;
+    testSubject(options, Fastly, util, testCacheKey);
+    util.serviceIdPresent.should.be.calledOnce();
+    Fastly.should.not.be.called();
+    done();
+  });
+
```

Comment:
We should add a test case that validates that this function stops without a cache key. That test should fail, because the function does not stop without the cache key. So then we build that functionality into it until our test is satisfied.